### PR TITLE
kinematic variables and baseline cuts

### PR DESCRIPTION
**Pull request by YooSunYoung**
<subsystem MT2>
<kinematic variables>
deltaR_ll
lepton1+2 mass, pt, px, py
lepton1&2 mass, pt
deltaR_bb
bottom1+2 mass, pt, px, py
bottom1&2 mass, pt
bottom1+bottom2+lepton1+lepton2 mass
<truth matching variables>
mother & grandmother PID of leptons & bottoms
<baseline cut information>
step

### CODE REVIEW

File: analysis/bin/doubleHiggsAnalyser.cc [hunk 1]
```diff
@@ -31,34 +31,64 @@ GenParticle* getMother(TClonesArray* particles, GenParticle* p){
...
+int* isFrom(TClonesArray* particles, int ip){
    auto p = static_cast<GenParticle *>(particles->At(ip));
    // check if it's from Higgs
    auto mom = getMother(particles, p); 
    if (mom==nullptr) return 0;
    auto grmom = getMother(particles, mom);
    if (grmom==nullptr) return 0;
-   return grmom->PID;
+   static int pedigree[2] = {mom->PID, grmom->PID};
+   return pedigree;
```
Comment (jshlee) [inline analysis/bin/doubleHiggsAnalyser.cc:45]: std::pair would be better than pointer of int

File: analysis/bin/doubleHiggsAnalyser.cc [hunk 2]
```diff
@@ -82,135 +113,238 @@ void doubleHiggsAnalyser::Initiate(TString output_file_name) {
...
-  //map<Float_t, int, greater<Float_t>> muons : map of <pt,index>:<K,V> of muon sorted by pt.
+  //map<Float_t, int, greater<Float_t>> leptons : map of <pt,index>:<K,V> of muon sorted by pt.
   //base selections : MissingET > 20, pT(lepton) > 20, deltaR(ll) < 1.0, m(ll) < 65, deltaR(bb) < 1.3, 95 < m(bb) < 140
+    
+    // Missing ET
     auto m = static_cast<const MissingET *>(missings->At(0)); // There is always one MET object.
-    if (m->MET<20) return false;
-    missing.SetPtEtaPhiM(m->MET,0,m->Phi,0);
- 
+    missing.SetPtEtaPhiM(m->MET,m->Eta,m->Phi,0);
```
Comment (jshlee) [inline analysis/bin/doubleHiggsAnalyser.cc]: met has no eta

File: analysis/bin/doubleHiggsAnalyser.cc [hunk 3]
```diff
@@ -82,135 +113,238 @@ void doubleHiggsAnalyser::Initiate(TString output_file_name) {
...
+    Mt2::LorentzTransverseVector vis_B(Mt2::TwoVector(bottombottom.Px(), bottombottom.Py()), bottombottom.M());
     Mt2::TwoVector pT_Miss(missing.Px(), missing.Py());
     
     lester_MT2 = asymm_mt2_lester_bisect::get_mT2( // the simpliest way to calculate MT2
-              muon1.M(),muon1.Px(),muon1.Py(),
-              muon2.M(),muon2.Px(),muon2.Px(),
+              leptonlepton.M(),leptonlepton.Px(),leptonlepton.Py(),
+              bottombottom.M(),bottombottom.Px(),bottombottom.Px(),
               missing.Px(),missing.Py(),
               missing.M(),missing.M());
```
Comment (jshlee) [inline analysis/bin/doubleHiggsAnalyser.cc]: why does this function take the same input, missing.M(), twice?

File: analysis/bin/doubleHiggsAnalyser.cc (hunk 2 above)
Comment (YooSunYoung) [reply analysis/bin/doubleHiggsAnalyser.cc]: changed 'm->Eta' to '0'

File: analysis/bin/doubleHiggsAnalyser.cc (hunk 1 above)
Comment (YooSunYoung) [reply analysis/bin/doubleHiggsAnalyser.cc:45]: replaced pointer of int with std::pair

File: analysis/bin/doubleHiggsAnalyser.cc (hunk 3 above)
Comment (YooSunYoung) [reply analysis/bin/doubleHiggsAnalyser.cc]: the 'get_mT2' function in lester bisect header file looks like this and there is only one invisible mass... so... should I set them to 0 instead?

```
static double get_mT2( // returns asymmetric mT2 (which is >=0), or returns a negative number (such as MT2_ERROR) in the case of an error.
    const double mVis1, const double pxVis1, const double pyVis1,
    const double mVis2, const double pxVis2, const double pyVis2,
    const double pxMiss, const double pyMiss,
    const double mInvis1, const double mInvis2,
    const double desiredPrecisionOnMT2=0, // This must be non-negative.  If set to zero (default) MT2 will be calculated to the highest precision available on the machine (or as close to that as the algorithm permits).  If set to a positive value, MT2 (note that is MT2, not its square) will be calculated to within +- desiredPrecisionOnMT2. Note that by requesting precision of +- 0.01 GeV on an MT2 value of 100 GeV can result in speedups of a factor of ...
    const bool useDeciSectionsInitially=true // If true, interval is cut at the 10% point until first acceptance, which gives factor 3 increase in speed calculating kinematic min, but 3% slowdown for events in the bulk.  Is on (true) by default, but can be turned off by setting to false.
  )
```

**Comment by jshlee** [comment]
a lot of files have been deleted. is this on purpose?
@slowmoyang do u need these files?

File: README.md [hunk 4]
```diff
@@ -1,21 +1,13 @@
...
-=======
-
 ## How to git clone
 ```
-scram p -n delphys CMSSW CMSSW_9_4_4
-cd delphys/src
+scram p -n delPhys CMSSW CMSSW_9_4_4
+cd delPhys/src
 cmsenv
 git clone git@github.com:"your_user_name"/nano.git #"your_user_name" should be replaced by real user name
```
Comment (jshlee) [inline README.md]: git clone git@github.com:CPLUOS/nano.git

**Comment by jshlee** [comment]
for global vars like the ones that will be saved to root, use m_* or *_ to tell them apart
bottom_m is a bit misleading, i would rather m_bb_m, m_ll_m etc

File: README.md (hunk 4 above)
Comment (YooSunYoung) [reply README.md]: changed the typo

**Comment by slowmoyang** [comment]
> a lot of files have been deleted. is this on purpose?
> @slowmoyang do u need these files?

no I did't remove files... it seems wrong

**Comment by jshlee** [comment]
+1
im ok with these
@slowmoyang can u check this

**Comment by jshlee** [comment]
all done?

**Pull request merged**